### PR TITLE
Use Cloud build private pools

### LINKED ISSUE #681: Create CloudBuild Private Pools
### TL;DR

Size: M

### Terraform Resources

_No response_

### Detailed design

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by daniel-cit**
Fixes #681

### CODE REVIEW

File: 0-bootstrap/README.md [hunk 1]
```diff
@@ -111,6 +111,40 @@ All groups in the `groups.optional_groups` are optional.
...
+
+
+   ```bash
+   export project_id=<ENV_SECRETS_PROJECT>
+   export secrete_name=<VPN_PSK_SECRET_NAME>
+   export member="serviceAccount:<BOOTSTRAP_TERRAFORM_SERVICE_ACCOUNT>|user:<YOUR_EMAIL>"
+
+   echo '<YOUR-PRESHARED-KEY-SECRET>' | gcloud secrets create "${secrete_name}" --project "${project_id}" --replication-policy=automatic --data-file=-
+
+   gcloud secrets add-iam-policy-binding "${secrete_name}"  --member="${member}" --role='roles/secretmanager.viewer' --project "${project_id}"
```
Comment (bharathkkb) [inline 0-bootstrap/README.md:141]:
```suggestion
   gcloud secrets add-iam-policy-binding "${secret_name}"  --member="${member}" --role='roles/secretmanager.viewer' --project "${project_id}"
```

File: 0-bootstrap/README.md [hunk 2]
```diff
@@ -111,6 +111,40 @@ All groups in the `groups.optional_groups` are optional.
...
+
+   gcloud secrets add-iam-policy-binding "${secrete_name}"  --member="${member}"  --role='roles/secretmanager.secretAccessor' --project "${project_id}"
```
Comment (bharathkkb) [inline 0-bootstrap/README.md:143]:
```suggestion
   gcloud secrets add-iam-policy-binding "${secret_name}"  --member="${member}"  --role='roles/secretmanager.secretAccessor' --project "${project_id}"
```

File: 0-bootstrap/modules/cb-private-pool/variables.tf [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+ */
+
+variable "project_id" {
+  description = "ID of the project where the private pool will be created"
+  type        = string
+}
+
+variable "private_worker_pool" {
+  description = <<EOT
+  name: Name of the worker pool.
```
Comment (bharathkkb) [inline 0-bootstrap/modules/cb-private-pool/variables.tf:24]: nit:
```suggestion
  name: Name of the worker pool. A name with a random suffix is generated if not set.
```

**Review by bharathkkb — APPROVED** [review]

File: 0-bootstrap/README.md (hunk 2 above)
Comment (daniel-cit) [reply 0-bootstrap/README.md]: fixed

Comment (daniel-cit) [reply 0-bootstrap/README.md]: fixed

File: 0-bootstrap/README.md [hunk 4]
```diff
@@ -111,6 +111,40 @@ All groups in the `groups.optional_groups` are optional.
...
+
+1. Create a secret for the VPN private pre-shared key and grant required roles to the identity used for the deploy, your user email or the Bootstrap terraform service account.
+
+
+   ```bash
+   export project_id=<ENV_SECRETS_PROJECT>
+   export secrete_name=<VPN_PSK_SECRET_NAME>
+   export member="serviceAccount:<BOOTSTRAP_TERRAFORM_SERVICE_ACCOUNT>|user:<YOUR_EMAIL>"
+
+   echo '<YOUR-PRESHARED-KEY-SECRET>' | gcloud secrets create "${secrete_name}" --project "${project_id}" --replication-policy=automatic --data-file=-
```
[the remark this answers is not in the document]
Comment (daniel-cit) [reply 0-bootstrap/README.md]: fixed

File: 0-bootstrap/README.md (hunk 1 above)
Comment (daniel-cit) [reply 0-bootstrap/README.md]: fixed

File: 0-bootstrap/README.md [hunk 5]
```diff
@@ -111,6 +111,40 @@ All groups in the `groups.optional_groups` are optional.
...
+
+0-bootstrap step also has an optional High Availability VPN configuration that can be used to on-prem connection.
+To enable this configuration do the following steps
+
+1. Create a secret for the VPN private pre-shared key and grant required roles to the identity used for the deploy, your user email or the Bootstrap terraform service account.
+
+
+   ```bash
+   export project_id=<ENV_SECRETS_PROJECT>
+   export secrete_name=<VPN_PSK_SECRET_NAME>
```
[the remark this answers is not in the document]
Comment (daniel-cit) [reply 0-bootstrap/README.md]: fixed

File: 0-bootstrap/modules/cb-private-pool/variables.tf (hunk 3 above)
Comment (daniel-cit) [reply 0-bootstrap/modules/cb-private-pool/variables.tf]: description updated

File: 0-bootstrap/README.md [hunk 6]
```diff
@@ -111,6 +111,40 @@ All groups in the `groups.optional_groups` are optional.
 
 Please refer to [troubleshooting](../docs/TROUBLESHOOTING.md) if you run into issues during this step.
 
+## Cloud Build access to on-prem
```
[the remark this answers is not in the document]
Comment (daniel-cit) [reply 0-bootstrap/README.md]: instructions moved to  `0-bootstrap/onprem.md`

**Pull request merged**